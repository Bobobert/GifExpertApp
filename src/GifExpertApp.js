import React, {useState} from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";


export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     //setCategories(categories.concat(['algo']));
    //     setCategories( ['Algo', ...categories] ); // con el uso del spread
    // }

    // return <>
    //   <h2>GifExpertApp</h2>
    //   <AddCategory setCategories={ setCategories }/>
    //   <hr/>

    //     <button> Agregar </button>

    //   <ol>
    //       {
    //           categories.map( category => {
    //               return <li key={category}> {category} </li>
    //           })
    //       }
    //   </ol>
    // </>;

    return <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr/>

        {/* <button> Agregar </button> */}

      <ol>
          {
              categories.map( category => 
                (<GifGrid 
                        key={ category }
                        category={category} 
                />))
          }
      </ol>
    </>;
  };