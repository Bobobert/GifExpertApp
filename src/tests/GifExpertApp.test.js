import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../GifExpertApp"

describe('Pruebas sobre el <GifExpertApp />', () => {

    test('debe mostrarse de manera correcta', () => {
        const {container} = render(<GifExpertApp defaultCategories={['Hallo']} />);
        expect( container ).toMatchSnapshot();
    })

    test('debe mostrar categorias correctamente', async() => {

        const categories = ['master chief', 'halo kitty'];

        render(<GifExpertApp defaultCategories={categories} />);

        // screen.findAllByTestId('GifGrid').then( recieved => {
        //     expect( recieved.length ).toBe( categories.length )
        // });

        expect( await screen.findAllByTestId('GifGrid') )
            .toHaveLength( categories.length );
    })

})