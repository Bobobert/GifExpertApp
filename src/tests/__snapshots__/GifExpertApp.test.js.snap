// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <GifExpertApp /> debe mostrarse de manera correcta 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <form>
    <input
      type="text"
      value=""
    />
  </form>
  <hr />
  <ol>
    <h3>
       
      Hallo
       
    </h3>
    <p
      class="animate__animated animate__shakeY"
    >
      Cargando . . .
    </p>
    <div
      class="card-grid"
      data-testid="GifGrid"
    />
  </ol>
</div>
`;
